packages/cron/src/jobs/metrics.js

```diff
@@ -85,6 +91,25 @@ async function updateUploadsCount(roPg, rwPg, type) {
   ])
 }
 
+/**
+ * @param {Client} roPg
+ * @param {Client} rwPg
+ */
+async function updateUploadTotal(roPg, rwPg) {
+  const { rows } = await roPg.query(COUNT_TOTAL_UPLOADS)
+  if (!rows.length)
+    throw new Error(`no rows returned counting past weeks uploads`)
+  await rwPg.query(UPDATE_METRIC, [`uploads_all_types_total`, rows[0].total])
+}
+
+/**
+ * @param {Client} roPg
+ * @param {Client} rwPg
+ */
+async function updateUploadStats(roPg, rwPg) {
+  await roPg.query(REFRESH_UPLOAD_STATS)
```

Comment:
I'm not sure about having a separate materialized view here. Why not just calculate each metric concurrently and insert/update it into the metrics table like the rest? What do we gain by having a materialized view?

Reply:
The logic for the materialized view is all held in SQL, it feels much easier to reason about, change, update and refresh _only_ the stats we return for the website, vs having separate individual metrics that get combined server side. There's no meaningful difference outside of not having to maintain several independent queries.

I'm super open, I've got a handful of alternative solutions, some of them more performant but less accurate, if your think adding them directly the metrics table with no materialized view is the way to go, I can do that.

Reply:
I rewrote this to use the existing metrics table. We don't have to count the total again, if we already query each upload type individually, I can just sum that amount as the total and diff the past 7 days on the frontend.

I removed the materialized view and brought all the additional metrics queries into the metrics table + cron